app/code/Oleksandrr/Chat/Controller/Message/Messages.php

```diff
@@ -19,35 +20,56 @@ class Messages extends \Magento\Framework\App\Action\Action implements
      */
     private $logger;
 
+    /**
+     * @var \Magento\Customer\Model\Session
+     */
+    private $userSession;
+
     /**
      * Save constructor.
      * @param \Magento\Framework\App\Action\Context $context
      * @param \Oleksandrr\Chat\Model\ResourceModel\Message\CollectionFactory $messageCollectionFactory
      * @param \Psr\Log\LoggerInterface $logger
+     * @param \Magento\Customer\Model\Session $userSession
      */
     public function __construct(
         \Magento\Framework\App\Action\Context $context,
         \Oleksandrr\Chat\Model\ResourceModel\Message\CollectionFactory $messageCollectionFactory,
-        \Psr\Log\LoggerInterface $logger
+        \Psr\Log\LoggerInterface $logger,
+        \Magento\Customer\Model\Session $userSession
     ) {
         parent::__construct($context);
         $this->messageCollectionFactory = $messageCollectionFactory;
         $this->logger = $logger;
+        $this->userSession = $userSession;
     }
 
     public function execute()
     {
         $collection = [];
+        $message = '';
 
         try {
-            $messageCollection = $this->messageCollectionFactory->create();
-            $messageCollection
-                ->addFieldToSelect('author_name')
-                ->addFieldToSelect('message')
-                ->addFieldToSelect('created_at')
-                ->setPageSize(10);
-            $collection = $messageCollection->getData();
-            $message = 'Post count: ' . count($collection);
+            if ($this->userSession->isLoggedIn()) {
+                $messageCollection = $this->messageCollectionFactory->create()->addAuthorFilter((int) $this->userSession->getId());
```

Comment:
@pandaUbuntu Should filter collection by chat hash, not customerId - otherwise you do not get admin answers. Collection is anyway filtered by chat hash - both for guests and registered customers.

By the way, use the word `customer` (e.g. buyer) for storefront visitors, not `user`. People with access to Admin Panel are `users`